src/cloudflare/internal/vectorize-api.ts

```diff
@@ -1,7 +1,14 @@
 // Copyright (c) 2023 Cloudflare, Inc.
 // Licensed under the Apache 2.0 license found in the LICENSE file or at:
 //     https://opensource.org/licenses/Apache-2.0
-import { default as flags } from 'workerd:compatibility-flags';
+declare global {
+  const Cloudflare: {
+    readonly compatibilityFlags: Record<string, boolean>;
+  };
+}
+
+const queryMetadataOptional =
+  !!Cloudflare.compatibilityFlags['vectorize_query_metadata_optional'];
```

Comment:
Why do we need !!? The type says it's already a boolean?

Reply:
Not strictly needed but I prefer it.